webapp/javascript/components/ComparisonApp.jsx

```diff
@@ -24,6 +24,51 @@ function ComparisonApp(props) {
     props;
   const { rawLeft, rawRight } = comparison;
 
+  const [linkedSearch, setLinkedSearch] = useState({
+    isSearchLinked: false,
+    linkedSearchQuery: '',
+    resetLinkedSearchSide: '',
+  });
+
+  const setSearchQuery = (query) => {
+    setLinkedSearch((x) => {
+      return {
+        ...x,
+        linkedSearchQuery: query,
+      };
+    });
+  };
+  const toggleLinkedSearch = (side) => {
+    const { isSearchLinked } = linkedSearch;
+    if (isSearchLinked) {
+      if (side === 'left') {
```

Comment:
Can we use `switch/case` here? This file isn't using typescript yet, but in the future we can update it to use exhaustive case checks.

Reply:
Plus these cases are pretty similar, so we can just build an `args` parameter and then declaratively use it?

Eg

```js
let args;
if (side === 'right') {
  args = { resetLinkedSearchSide: 'left' }
}
if (side === 'both') {
  args = { sSearchLinked: false, resetLinkedSearchSide: '' }
}
(...)

setLinkedSearch((x) => {
   return { ...x,  ...args };
});
```

Reply:
Also I think having all this logic here bloats this file. These files (Coparison, SingleView, ComparisonDiff) are the entrypoints for our pages, ie the first or second thing people should read to understand what the page is about. Having this logic makes it harder for people to understand it.

Reply:
@eh-am where would you put this instead?

Reply:
> Can we use switch/case here
> declarative way

done.